CMakeLists.txt

```diff
@@ -95,87 +80,151 @@ add_library(
 	cuda-api-wrappers
 	src/cuda/api/device_properties.cpp
 	src/cuda/api/profiling.cpp
-	src/cuda/api/ipc.hpp
-	src/cuda/api/event.hpp
-	src/cuda/api/device_properties.hpp
-	src/cuda/api/pointer.hpp
-	src/cuda/api/multi_wrapper_impls.hpp
-	src/cuda/api/current_device.hpp
-	src/cuda/api/peer_to_peer.hpp
-	src/cuda/api/device.hpp
-	src/cuda/api/pci_id.hpp
-	src/cuda/api/versions.hpp
-	src/cuda/api/unique_ptr.hpp
-	src/cuda/api/memory.hpp
-	src/cuda/api/device_count.hpp
-	src/cuda/api/stream.hpp
-	src/cuda/api/error.hpp
-	src/cuda/api/device_function.hpp
-	src/cuda/api/miscellany.hpp
 )
 
+# CUDA_STANDARD can not be set as `compile_feature` yet. Thus we
+# set the c++ standard, since that is used in the header part of
+# this library, to force users of this library to compile automatically
+# with c++11 compatible standards.
+set_target_properties(cuda-api-wrappers PROPERTIES CUDA_STANDARD 11)
+target_compile_features(cuda-api-wrappers PUBLIC cxx_std_11)
+
+target_include_directories(
+	cuda-api-wrappers
+	PUBLIC
+		"$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>"
+		"$<INSTALL_INTERFACE:include>"
+)
+
+target_link_libraries(cuda-api-wrappers cudart)
+
+# To enable `target_link_library(foo cuda-api-wrappers::cuda-api-wrappers)`
+# in other CMake projects, which include this project with `add_subdirectory`
+add_library(cuda-api-wrappers::cuda-api-wrappers ALIAS cuda-api-wrappers)
+
 # -----------------------
 # Examples / Tests
 # -----------------------
 
-find_library(CUDART_LIBRARY cudart ${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES})
-link_libraries(cuda-api-wrappers ${CUDART_LIBRARY})
```

Comment:
How are we going to get the libraries for the executables which aren't compiled by nvcc? We need this (well, actually, we need `${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES}`, see my most recent commit).

Reply:
Enabling the CUDA language automatically finds all CUDA libraries for me. Does'nt this work for you? Basically I kept line 122. It is now `link_libraries(cuda-api-wrappers cudart)` in line 99.

Reply:
No, it doesn't find them. That is, not when I'm compiling regular .cpp files with gcc/clang. But I've rewritten this to use a more appropriate variable.

Reply:
That is true. I think it is reasonable to have the CUDA libraries in your `LD_LIBRARY_PATH`, because there are a lot of systems, which have multiple CUDA versions installed. Some systems use environment modules to decide which CUDA version to use. Moreover CUDA can be installed in a custom path. If I remove now my CUDA libraries from `LD_LIBRARY_PATH` how does CMake know, which version I want to use -- It cannot. Especially if you think about modern system-independent package managers (e.g. `spack`), it is reasonable to have the CUDA libraries in your `LD_LIBRARY_PATH`.

As a note: providing a `cuda-api-wrappers` installation via `spack` would increase user base and useability of your project. You might be interested in that.

Reply:
But we definitely omit the target-based approach here. You are right. So defining `cudart` library as a seperate target, should be the clean solution.